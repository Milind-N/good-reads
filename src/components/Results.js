import React from 'react'
import PropTypes from 'prop-types'
import SearchResult from './SearchResult'

const Results = ({ books, expandBook }) => {

  return (
    <div>
      <div className="row">
        {(books.length > 0 ) ? <p>PAGE 1 OF ABOUT {books.length} RESULTS (0.18 SECONDS)</p> : ""}
      </div>
      <div className="row">
        {books.map(book => (
          <SearchResult books={book} key={book.id} expandBook={expandBook} />
        ))
        }
      </div>
    </div>

  )
}

Results.propTypes = {
  books: PropTypes.array,
  expandBook: PropTypes.func
}

export default Results
