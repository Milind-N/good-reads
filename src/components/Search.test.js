import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from './Search';
import Results from './Results';

describe('Search renders correctly', () => {
  let props;

  beforeEach(() => {
    props = {
      books: [{
            		"average_rating": "4.46",
            		"best_book": {
            			"author": {
            				"id": "1077326",
            				"name": "J.K. Rowling"
            			},
            			"id": "3",
            			"image_url": "https://images.gr-assets.com/books/1474154022m/3.jpg",
            			"small_image_url": "https://images.gr-assets.com/books/1474154022s/3.jpg",
            			"title": "Harry Potter and the Sorcerer's Stone (Harry Potter, #1)"
            		},
            		"books_count": "621",
            		"id": "4640799",
            		"original_publication_day": "26",
            		"original_publication_month": "6",
            		"original_publication_year": "1997",
            		"ratings_count": "5620117",
            		"text_reviews_count": "89468"
            	},
            	{
            		"average_rating": "4.55",
            		"best_book": {
            			"author": {
            				"id": "1077326",
            				"name": "J.K. Rowling"
            			},
            			"id": "5",
            			"image_url": "https://images.gr-assets.com/books/1499277281m/5.jpg",
            			"small_image_url": "https://images.gr-assets.com/books/1499277281s/5.jpg",
            			"title": "Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)"
            		},
            		"books_count": "440",
            		"id": "2402163",
            		"original_publication_day": "8",
            		"original_publication_month": "7",
            		"original_publication_year": "1999",
            		"ratings_count": "2228376",
            		"text_reviews_count": "43197"
            	}
            ],
      expandBook: jest.fn(() => 'expandBook')
    };
  });

  const wrapper = shallow(<Search />);

  it('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders one input text field', () => {
    expect(wrapper.find("input")).toHaveLength(1);
  });

  it('renders one button for search', () => {
    expect(wrapper.find("button")).toHaveLength(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Results {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

})
