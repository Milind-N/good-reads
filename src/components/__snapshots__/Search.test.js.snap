// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <p>
      PAGE 1 OF ABOUT 
      2
       RESULTS (0.18 SECONDS)
    </p>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-2 col-sm-4 col-md-3"
    >
      <div
        className="card"
      >
        <img
          alt="Book cover"
          className="card-img-top pl-2 pr-2 pt-2"
          height="200px"
          onClick={[Function]}
          src="https://images.gr-assets.com/books/1474154022m/3.jpg"
        />
        <div
          className="card-body"
        >
          <p
            className="text-sm-left card-title "
            data-placement="bottom"
            data-toggle="tooltip"
            title="Harry Potter and the Sorcerer's Stone (Harry Potter, #1)"
          >
            Harry Potter and the...
          </p>
          <p
            className="text-sm-left card-text"
          >
            J.K. Rowling
          </p>
        </div>
      </div>
    </div>
    <div
      className="col-lg-2 col-sm-4 col-md-3"
    >
      <div
        className="card"
      >
        <img
          alt="Book cover"
          className="card-img-top pl-2 pr-2 pt-2"
          height="200px"
          onClick={[Function]}
          src="https://images.gr-assets.com/books/1499277281m/5.jpg"
        />
        <div
          className="card-body"
        >
          <p
            className="text-sm-left card-title "
            data-placement="bottom"
            data-toggle="tooltip"
            title="Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)"
          >
            Harry Potter and the...
          </p>
          <p
            className="text-sm-left card-text"
          >
            J.K. Rowling
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
